allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -816,43 +816,16 @@ def step_count(self, val: int):
     def log_interval(self):
         return self.training_pipeline.metric_accumulate_interval
 
-    def act(self, rollouts: RolloutStorage):
+    def act(self, rollouts: RolloutStorage, dist_wrapper_class: Optional[type] = None):
+        dist_wrapper_class = None
+        if self.training_pipeline.current_stage.teacher_forcing is not None:
+            dist_wrapper_class = TeacherForcingDistr
```

Comment:
Yes, I see. Currently (if I'm not wrong), I'm still outputting the 0.25 of the model for the chosen action while enforcing actions during sampling (so `TeacherForcingDistr` does not fully behave as a distribution). Maybe we want to call it by a different name? It's used as a distribution. Maybe we need a superclass for `Distr` called `Sampleable`, which can specialize as a `Distr` (with all its expected properties) or a `Teacher`, which kind of hacks some details since it's no longer a `Distr`?